packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
 
   /**
    * List of enabled authentication flows
-   * @default no enabled flows
+   * @default - no enabled flows
    */
   readonly enabledAuthFlows?: AuthFlow[]
 }
 
 /**
- * Define a UserPool App Client
+ * Represents a Cognito user pool client.
  */
-export class UserPoolClient extends Resource {
+export interface IUserPoolClient extends IResource {
   /**
+   * Name of the application client
    * @attribute
    */
-  public readonly userPoolClientId: string;
+  readonly userPoolClientId: string;
+}
 
+/**
+ * Define a UserPool App Client
+ */
+export class UserPoolClient extends Resource implements IUserPoolClient {
   /**
-   * @attribute
+   * Import a user pool client given its id.
    */
-  public readonly userPoolClientName: string;
+  public static fromUserPoolClientId(scope: Construct, id: string, userPoolClientId: string): IUserPoolClient {
+    class Import extends Resource implements IUserPoolClient {
+      public readonly userPoolClientId = userPoolClientId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly userPoolClientId: string;
 
   /**
-   * @attribute
+   * The client name that was specified via the `userPoolClientName` property during initialization,
+   * `undefined` otherwise.
+   */
+  public readonly userPoolClientName?: string;
+
+  /*
+   * Note to implementers: Two CloudFormation return values Name and ClientSecret are part of the spec.
+   * However, they have been explicity not implemented here. They are not documented in CloudFormation, and
+   * CloudFormation returns the following the string when these two attributes are 'GetAtt' - "attribute not supported
+   * at this time, please use the CLI or Console to retrieve this value".
+   * Awaiting updates from CloudFormation.
    */
-  public readonly userPoolClientClientSecret: string;
 
   constructor(scope: Construct, id: string, props: UserPoolClientProps) {
-    super(scope, id, {
-      physicalName: props.userPoolClientName,
-    });
+    super(scope, id);
```

Comment:
Why are we not using physical name?

Reply:
User pool clients generate their own primary identifiers (e.g. `6tapehhff4ov922cjnheqpfr23`) that cannot be specified as input. AFAIU, physical names are set for resources whose primary identifiers can be configured by the user. Am I wrong?

Reply:
Okay makes sense. Maybe add a comment.